py/modbuiltins.c

```diff
@@ -440,8 +440,18 @@ STATIC mp_obj_t mp_builtin_round(mp_uint_t n_args, const mp_obj_t *args) {
     mp_int_t num_dig = 0;
     if (n_args > 1) {
         num_dig = mp_obj_get_int(args[1]);
-        if (num_dig > 0) {
-            mp_not_implemented("round(..., N>0)");
+        if (num_dig > 0 && num_dig < 7) {
+            mp_float_t val = mp_obj_get_float(o_in);
+            mp_float_t mult = MICROPY_FLOAT_C_FUN(pow)(10, num_dig);
+            mp_float_t rounded = MICROPY_FLOAT_C_FUN(round)(val * mult) / mult;
+            return mp_obj_new_float(rounded);
+        } else if (num_dig > 0 && num_dig > 6) {
+            mp_not_implemented("round(..., N>6)");
+        } else if (num_dig < 0) {
+            mp_float_t val = mp_obj_get_float(o_in);
+            mp_float_t mult = MICROPY_FLOAT_C_FUN(pow)(10, MICROPY_FLOAT_C_FUN(fabs)(num_dig));
+            mp_float_t rounded = MICROPY_FLOAT_C_FUN(round)(val * mult);
```

Comment:
I don't think this implementation is correct. I'm still -0.3 on not implementing < 0 case at all (how often it is used? ~never).